File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0; 
        Arrays.sort(nums);
        int max = 0;
        int curr = 1;
        for(int i = 0 ; i < nums.length - 1 ; i++){
            if(nums[i]+1 != nums[i+1] && nums[i] != nums[i+1]){
                max = Math.max(max , curr);
                curr = 1;
            }else if(nums[i] != nums[i+1]){
                curr++;
            }
        }
        max = Math.max(max , curr);
        return max;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashset = set(nums)
        ans = 0
        for num in hashset:
            if not num - 1 in hashset:
                length = 1
                while(length + num in hashset):
                    length += 1  
                ans = max(ans, length)
        return ans 
                